projects/storefrontapp-e2e-cypress/cypress/helpers/asm.ts

```diff
@@ -222,5 +207,51 @@ export function listenForUserDetailsRequest(): string {
   const aliasName = 'userDetails';
   cy.server();
   cy.route('GET', '/rest/v2/electronics-spa/users/*').as(aliasName);
+  cy.log(aliasName);
   return `@${aliasName}`;
 }
+
+export const agentAuthentication = () => {
+  const aliasRequest = asm.listenForAuthenticationRequest();
+
+  cy.get('cx-csagent-login-form').should('exist');
+  cy.get('cx-customer-selection').should('not.exist');
+  cy.get('cx-csagent-login-form form').within(() => {
+    cy.get('[formcontrolname="userId"]').type('asagent');
+    cy.get('[formcontrolname="password"]').type('123456');
+    cy.get('button[type="submit"]').click();
+  });
+
+  cy.wait(aliasRequest)
+    .its('status')
+    .should('eq', 200);
+  cy.get('cx-csagent-login-form').should('not.exist');
+  cy.get('cx-customer-selection').should('exist');
+};
+
+export const customerAuthentication = fullname => {
```

Comment:
Lets discuss the full name parameter and the function syntax.  Also, this was probably added by me previously, but there is something fishy about using the asm. prefix to call functions.